torch/csrc/jit/codegen/cuda/ir_builder.h

```diff
@@ -9,17 +9,26 @@ namespace jit {
 namespace fuser {
 namespace cuda {
 
+namespace kir {
+class Kernel;
+}
+
 class IrCloner;
 
 // Passkey for builder to register properties with statements, and to call
 // functions in IrContainer
 class TORCH_CUDA_CU_API IrBuilderPasskey {
   friend class IrBuilder;
+  friend class kir::IrBuilder;
 
  public:
+  // TODO: Collapse ir_container and Kernel once Kernel inherits from
+  // IrContainer
   IrContainer* const ir_container_ = nullptr;
+  kir::Kernel* const kernel = nullptr;
 
  private:
+  explicit IrBuilderPasskey(kir::Kernel* kernel);
```

Comment:
nit: any particular reason this constructor is defined externally, whereas the `IrContainer*` version is inlined?

Reply:
I want to keep the create definitions in this file which require the fusion guard and ir container definitions, however Kernel isn't otherwise used, so kept the include dependency in the .cpp file. I'm mostly trying to keep the dependencies as light as possible, but other than that I don't recall any specific reason.

Reply:
I think all_nodes could be removed too.